### PR TITLE
Calendar

### PR DESCRIPTION
**Pull request by bbalbon**

### CODE REVIEW

File: client/src/components/calendar.jsx [hunk 1]
```diff
@@ -1,20 +1,64 @@
 import React from 'react';
+import moment from 'moment';
+import Month from './month.jsx';
 
 class Calendar extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
-            current: null
+            current: moment().month(),
```
Comment (bbalbon) [inline client/src/components/calendar.jsx]: Here I'm delegating the current month to be an integer out of 11

File: client/src/components/calendar.jsx [hunk 2]
```diff
@@ -1,20 +1,64 @@
...
+    getDaysInMonth (month) {
+        return moment().month(month).daysInMonth();
+    }
+
+    advanceMonth () {
+        const newMonth = this.state.current + 1;
+        this.setState({
+            current: newMonth,
+            first: moment.months()[newMonth],
+            second: moment.months()[newMonth + 1]
```
Comment (bbalbon) [inline client/src/components/calendar.jsx]: This function increments the currentMonth counter, automatically triggering the re-render with the correct month (old + 1)

File: client/src/components/calendar.jsx [hunk 3]
```diff
@@ -1,20 +1,64 @@
...
+        })
+    }
+
+    previousMonth () {
+        const newMonth = this.state.current - 1;
+        this.setState({
+            current: newMonth,
+            first: moment.months()[newMonth],
+            second: moment.months()[newMonth + 1]
+        })
```
Comment (bbalbon) [inline client/src/components/calendar.jsx]: The logic here is identical to the advanceMonth function

File: client/src/components/month.jsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+class Month extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            current: null
+        }
+    }
+
+    getDaysFormatted () {
```
Comment (bbalbon) [inline client/src/components/month.jsx]: This function formats the days in the correct sequence based on the month I' passing in . The two most import function from moment are the "get # of days in month" and "get starting day of month"

**Comment by bbalbon** [comment]
Implemented the majority of the calendar logic. The rest is just CSS

File: client/src/components/calendar.jsx (hunk 2 above)
Comment (yavorona) [reply client/src/components/calendar.jsx]: Nice!

**Pull request merged**